nuxeo-retention/src/main/java/org/nuxeo/retention/service/RetentionManagerImpl.java

```diff
@@ -215,14 +216,12 @@ public void fireRetentionEvent(String eventName, String eventInput, boolean audi
         Event event = evctx.newEvent(eventName);
         Framework.getService(EventProducer.class).fireEvent(event);
         if (audit) {
-            AuditLogger logger = Framework.getService(AuditLogger.class);
-            LogEntry entry = logger.newLogEntry();
-            entry.setEventId(name);
-            entry.setEventDate(new Date());
-            entry.setCategory(RetentionConstants.EVENT_CATEGORY);
-            entry.setPrincipalName(session.getPrincipal().getName());
-            entry.setComment(evctx.getInput());
-            logger.addLogEntries(Collections.singletonList(entry));
+            LogEntry entry = LogEntry.builder(name, new Date())
+                                     .category(RetentionConstants.EVENT_CATEGORY)
+                                     .principalName(session.getPrincipal().getName())
+                                     .comment(evctx.getInput())
+                                     .build();
+            Framework.getService(AuditBackend.class).addLogEntries(Collections.singletonList(entry));
```

Comment:
This is not a good idea (for all Nuxeo versions), this should be done by the audit itself by contributing to `"org.nuxeo.audit.service.AuditComponent#event".

Reply:
We don't know in advance the eventId, it is a business one fired by 3rd party